### ISSUE TITLE
Move Identity Cookie code into separate functions in web\User.php

### ISSUE THREAD

**Issue by maine-mike** [yiisoft/yii2#8795]
Hi,

I tried this last week, but thought it might be better to start over.  Here is a pull request to separate out the cookie-related functions in web\User.php.  This will allow someone to write different cookie handling functions to override the base remember-me cookie functions.

https://github.com/yiisoft/yii2/pull/8714

Thank you!

**Comment by SilverFire** [comment]
Movin to 2.0.8 because the original PR was declined.

**Comment by maine-mike** [comment]
My original enhancement included a bug fix.  It appears that it's a bad idea to mix a bug fix with an enhancement.  So I created this pull request for the bug fix:

https://github.com/yiisoft/yii2/pull/10480

Once that is complete, I'll create a new pull request for this enhancement.  Thank you.

**Comment by samdark** [comment]
@maine-mike #10480 is now merged. Sorry that it took so long.

**Comment by SilverFire** [comment]
Resolved by https://github.com/yiisoft/yii2/commit/7249a6c99e96cafae522da379fc7c609ce910bc3 with minor adjustments in https://github.com/yiisoft/yii2/commit/38be74446e943f238f13be02ac93f1880451548c
